### PR TITLE
Distribution: Add support for Linux Mint 18.3

### LINKED ISSUE #52210: wrong format for ansible_distribution_version on Linux Mint 18.3
##### SUMMARY
ansible_distribution_version is stored in bad format (trailing double-quotes) and also ansible_distribution_major_version is wrong (it seems to be a consequence)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module setup

##### ANSIBLE VERSION

```
ansible 2.7.7
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/nicolas/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 12 2018, 14:36:49) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

```
ANSIBLE_NOCOWS(/etc/ansible/ansible.cfg) = True
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 5
DEFAULT_LOG_PATH(/etc/ansible/ansible.cfg) = /var/log/ansible/ansible.log
DEFAULT_REMOTE_USER(/etc/ansible/ansible.cfg) = root
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 10
INVENTORY_ENABLED(/etc/ansible/ansible.cfg) = [u'ini', u'host_list', u'virtualbox', u'yaml', u'constructed']

```

##### OS / ENVIRONMENT

```
$ ls -l /etc/os-release 
lrwxrwxrwx 1 root root 21 nov.   4  2017 /etc/os-release -> ../usr/lib/os-release

$ cat /usr/lib/os-release 
NAME="Linux Mint"
VERSION="18.3 (Sylvia)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 18.3"
VERSION_ID="18.3"
HOME_URL="http://www.linuxmint.com/"
SUPPORT_URL="http://forums.linuxmint.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/linuxmint/"
VERSION_CODENAME=sylvia
UBUNTU_CODENAME=xenial

$ lsb_release -a
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 18.3 Sylvia
Release:	18.3
Codename:	sylvia

```

##### STEPS TO REPRODUCE

I am checking the ansible_distribution_version in a when statement 
```
- name: Restart xymon-client
  become: true
  service:
    name: xymon-client
    state: restarted
  when:
    (__upgrade.changed or restarted_services.changed) and
    (
    (ansible_distribution == "Debian" and
    ansible_distribution_major_version|int >= 8) or
    (ansible_distribution == "Linux Mint" and
    ansible_distribution_major_version|int >= 17) or
    (ansible_distribution == "Ubuntu" and
    ansible_distribution_major_version|int >= 14)
    )
```
Since format is bad for Linux Mint (trailing double-quotes), the test is always failed

```
ansible -m setup localhost | grep ansible_distribution
        "ansible_distribution": "Linux Mint", 
        "ansible_distribution_file_parsed": true, 
        "ansible_distribution_file_path": "/usr/lib/os-release", 
        "ansible_distribution_file_variety": "ClearLinux", 
        "ansible_distribution_major_version": "\"18.3\"", 
        "ansible_distribution_release": "linuxmint", 
        "ansible_distribution_version": "\"18.3\"", 
```

##### EXPECTED RESULTS

```
"ansible_distribution_major_version": "18"
"ansible_distribution_version": "18.3"
```

##### ACTUAL RESULTS

```
"ansible_distribution_version": "\"18.3\""
"ansible_distribution_version": "\"18.3\""
```

```paste below

```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #52210

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/facts/system/distribution.py
test/units/module_utils/test_distribution_version.py

### CODE REVIEW

**Comment by maxamillion** [comment]
shipit

**Comment by maxamillion** [comment]
rebuild_merge

File: test/units/module_utils/test_distribution_version.py [hunk 1]
```diff
@@ -1157,6 +1157,24 @@
...
+    {
+        "platform.dist": [
+            "LinuxMint",
+            "18.3",
+            "sylvia",
+        ],
+        "input": {
+            "/etc/os-release": "NAME=\"Linux Mint\"\nVERSION=\"18.3 (Sylvia)\"\nID=linuxmint\nID_LIKE=ubuntu\nPRETTY_NAME=\"Linux Mint 18.3\"\nVERSION_ID=\"18.3\"\nHOME_URL=\"http://www.linuxmint.com/\"\nSUPPORT_URL=\"http://forums.linuxmint.com/\"\nBUG_REPORT_URL=\"http://bugs.launchpad.net/linuxmint/\"\nVERSION_CODENAME=sylvia\nUBUNTU_CODENAME=xenial",  # noqa
+        },
+        "name": "Linux Mint 18.03",
```
Comment (niko2) [inline test/units/module_utils/test_distribution_version.py]: 18.3

File: lib/ansible/module_utils/facts/system/distribution.py [hunk 2]
```diff
@@ -345,6 +345,12 @@ def parse_distribution_file_Debian(self, name, data, path, collected_facts):
             release = re.search(r'VERSION="(.*)"', data)
             if release:
                 debian_facts['distribution_release'] = release.groups()[0]
+        elif "Mint" in data:
+            debian_facts['distribution'] = 'Linux Mint'
+            version = re.search(r"VERSION_ID=\"(.*)\"", data)
+            if version:
+                debian_facts['distribution_version'] = version.group(1)
+                debian_facts['distribution_major_version'] = version.group(1)
```
Comment (niko2) [inline lib/ansible/module_utils/facts/system/distribution.py]: I think
`.split('.')[0]`
is needed to get exact major version

File: test/units/module_utils/test_distribution_version.py [hunk 3]
```diff
@@ -1157,6 +1157,24 @@
...
+        "result": {
+            "distribution_release": "sylvia",
+            "distribution": "Linux Mint",
+            "distribution_major_version": "18.3",
```
Comment (niko2) [inline test/units/module_utils/test_distribution_version.py]: "distribution_major_version": "18",

**Comment by Akasurde** [comment]
@niko2 Thanks for review comments. Could you please check once and let me know ? Thanks.

**Review by niko2 — APPROVED** [review]

**Comment by Akasurde** [comment]
@niko2 @maxamillion Thanks for the review.

**Pull request merged**